lib/fdo.c

```diff
@@ -1241,8 +1241,8 @@ static bool _STATE_TO2(void)
 			}
 			dns = g_fdo_data->current_rvto2addrentry->rvdns;
 			port = g_fdo_data->current_rvto2addrentry->rvport;
-			if (g_fdo_data->current_rvto2addrentry->rvprotocol == RVPROTHTTPS ||
-				g_fdo_data->current_rvto2addrentry->rvprotocol == RVPROTTLS) {
+			if (g_fdo_data->current_rvto2addrentry->rvprotocol == PROTHTTPS ||
```

Comment:
Has RVPROTHTTPS been removed completely? If not, it seems easy to confuse the two so is there comments explaining how and when they should be used?

Reply:
This is because there are two different set of values defined for the protocol configuration in RVInfo vs RVTO2Addr.

RVInfo Protocol values: section 3.7

RVProtocolValue /= (
    RVProtRest    => 0,
    RVProtHttp    => 1,
    RVProtHttps   => 2,
    RVProtTcp     => 3,
    RVProtTls     => 4,
    RVProtCoapTcp => 5,
    RVProtCoapUdp => 6
);

RVTO2Addr Protocol values: section 3.3.12 and 3.3.144
TransportProtocol /= (
    ProtTCP:    1,     ;; bare TCP stream
    ProtTLS:    2,     ;; bare TLS stream
    ProtHTTP:   3,
    ProtCoAP:   4,
    ProtHTTPS:  5,
    ProtCoAPS:  6,
)